### PR TITLE
[#17028] Tabbing through inline editor causes JavaScript error.

### LINKED ISSUE #424: Tabbing through inline editor causes JavaScript error
## Are you reporting a feature or a bug?

Bug

## [Check if the issue is already reported](http://docs.ckeditor.com/#!/guide/dev_issues_readme-section-avoid-duplicates)

http://dev.ckeditor.com/ticket/17028

## Provide detailed reproduction steps (if any)

1. Open inlineall.html or sharedspace.html sample
2. Click in one of editors
3. Start pressing tab

### Expected result

No errors get thrown.

### Actual result

```Cannot read property 'contains' of null``` is thrown once you leave last inline editor in sharedspace or after focusing third editor in inline all.

**Message:** Cannot read property 'contains' of null
**URL:** ckeditor-dev/plugins/indent/plugin.js
**Line:** 366

## Other details

Problem can be reproduced from CKEditor 4.6.2 only in Chrome.

* Browser: Chrome
* CKEditor version: 4.6.2

### PR DESCRIPTION
**Pull request by msamsel**
## What is the purpose of this pull request?

Bug fix
It's PR related to #424 

## Does your PR contain necessary tests?

yes

### This PR contains

- [x] Unit tests
- [x] Manual tests

## What changes did you make?

Add checks in 2 places that values are defined.

close #424

### CODE REVIEW

File: tests/plugins/indentlist/manual/taberrorininlineeditor.html [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<head>
+	<script>
+		if ( bender.tools.env.mobile ) {
+			bender.ignore;
```
Comment (Comandeer) [inline tests/plugins/indentlist/manual/taberrorininlineeditor.html]: `bender.ignore` is a function.

Comment (msamsel) [inline tests/plugins/indentlist/manual/taberrorininlineeditor.html]: Ups.
Done.

**Comment by msamsel** [comment]
It seems that problem start to occur after introducing change in 3f07b6d34f3ccab9527c25a74734f3a2ddab2753.
Problem describe there seems to be fixed in Chrome v57. I revert this change and make slightly modification in Unit test. As I see chrome completely remove selection so reading type throw an error. Now test check if native selection exists.

File: tests/plugins/indentlist/manual/taberrorininlineeditor.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+@bender-tags: 4.7.1, tc, gh424
```
Comment (Comandeer) [inline tests/plugins/indentlist/manual/taberrorininlineeditor.md]: Please update tags to the new convention.

File: tests/plugins/tab/manual/taberrorininlineeditor.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+@bender-tags: 4.7.1, tc, gh424
```
Comment (Comandeer) [inline tests/plugins/tab/manual/taberrorininlineeditor.md]: Please update tags to the new convention.

File: plugins/indentlist/plugin.js [hunk 4]
```diff
@@ -35,17 +35,24 @@
...
 				// be done for any list item that isn't the first child of the parent.
 				editor.on( 'key', function( evt ) {
+					var path = editor.elementPath();
+
 					if ( editor.mode != 'wysiwyg' )
 						return;
 
 					if ( evt.data.keyCode == this.indentKey ) {
-						var list = this.getContext( editor.elementPath() );
+						// Prevent of getting context of empty path. (#17028)
```
Comment (Comandeer) [inline plugins/indentlist/plugin.js]: Please update ticket number.

File: core/dom/elementpath.js [hunk 5]
```diff
@@ -57,6 +57,11 @@
 		// Backward compact.
 		root = root || startNode.getDocument().getBody();
 
+		// Assign root value if startNode is null (#17028).
```
Comment (Comandeer) [inline core/dom/elementpath.js]: Please update ticket number.

File: tests/core/focusmanager/editor.js [hunk 6]
```diff
@@ -109,7 +109,12 @@ bender.test( {
 			resume( function() {
 				editor.on( 'blur', function() {
 					resume( function() {
-						assert.areSame( 'None', editor.getSelection().getNative().type );
```
Comment (Comandeer) [inline tests/core/focusmanager/editor.js]: Are these tests still needed? It seems that now they're checking the native Chrome's implementation.

There is also manual test for it (`core/focusmanager/manual/typinginunfocusedinlineeditor`), which became unneeded.

File: core/dom/elementpath.js [hunk 7]
```diff
@@ -57,6 +57,11 @@
 		// Backward compact.
 		root = root || startNode.getDocument().getBody();
 
+		// Assign root value if startNode is null (https://dev.ckeditor.com/ticket/17028).
```
Comment (Comandeer) [inline core/dom/elementpath.js]: Update tickets numbers to the new convention.

File: plugins/indentlist/plugin.js [hunk 8]
```diff
@@ -35,17 +35,24 @@
...
 				// be done for any list item that isn't the first child of the parent.
 				editor.on( 'key', function( evt ) {
+					var path = editor.elementPath();
+
 					if ( editor.mode != 'wysiwyg' )
 						return;
 
 					if ( evt.data.keyCode == this.indentKey ) {
-						var list = this.getContext( editor.elementPath() );
+						// Prevent of getting context of empty path (https://dev.ckeditor.com/ticket/17028).
```
Comment (Comandeer) [inline plugins/indentlist/plugin.js]: Update tickets numbers to the new convention.

File: tests/plugins/indentlist/manual/taberrorininlineeditor.md [hunk 9]
```diff
@@ -0,0 +1,12 @@
+@bender-tags: 4.7.1, tc, 424
```
Comment (Comandeer) [inline tests/plugins/indentlist/manual/taberrorininlineeditor.md]: Update tags to the new bug/feature convention.

Additionally this test does not make much sense in browsers other than Chrome (as this thing was always working there) – let's run it only in Chrome.

File: tests/plugins/indentlist/taberror.js [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+
+(function () {
+	'use strict';
+
+	bender.editor = {
+		creator: 'inline'
+	};
+
+	bender.test({
+		'test tab in inline editor': function() {
```
Comment (Comandeer) [inline tests/plugins/indentlist/taberror.js]: Maybe add ticket reference here, just for clarity?

File: tests/plugins/tab/errorinlineeditor.js [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+
+(function () {
+	'use strict';
+
+	bender.editor = {
+		creator: 'inline'
+	};
+
+	bender.test({
+		'test tab in inline editor': function() {
```
Comment (Comandeer) [inline tests/plugins/tab/errorinlineeditor.js]: Maybe add ticket reference here, just for clarity?

File: tests/plugins/tab/manual/taberrorininlineeditor.md [hunk 12]
```diff
@@ -0,0 +1,12 @@
+@bender-tags: 4.7.1, tc, 424
```
Comment (Comandeer) [inline tests/plugins/tab/manual/taberrorininlineeditor.md]: Update tags to the new bug/feature convention.

Additionally this test does not make much sense in browsers other than Chrome (as this thing was always working there) – let's run it only in Chrome.

File: core/dom/elementpath.js (hunk 7 above)
Comment (msamsel) [inline core/dom/elementpath.js]: @Comandeer  I'm a little bit confused, according to #470, there should be permanent link to issue in track. And it is there, or do I missed something?
Or by new convention you understand reference to github issue?

Comment (Comandeer) [inline core/dom/elementpath.js]: Yes, I mean adding reference to GH issue alongside the reference to trac.

File: tests/plugins/indentlist/manual/taberrorininlineeditor.html [hunk 13]
```diff
@@ -0,0 +1,40 @@
+<head>
+	<script>
+		if ( !CKEDITOR.env.chrome ) {
```
Comment (Comandeer) [inline tests/plugins/indentlist/manual/taberrorininlineeditor.html]: Please move it to the `script` in `body` as it apparently does not work in `head`.

File: tests/plugins/tab/manual/taberrorininlineeditor.html [hunk 14]
```diff
@@ -0,0 +1,40 @@
+<head>
+	<script>
+		if ( !CKEDITOR.env.chrome ) {
```
Comment (Comandeer) [inline tests/plugins/tab/manual/taberrorininlineeditor.html]: Please move it to the `script` in `body` as it apparently does not work in `head`.

File: tests/plugins/indentlist/manual/taberrorininlineeditor.html (hunk 13 above)
Comment (msamsel) [inline tests/plugins/indentlist/manual/taberrorininlineeditor.html]: Interesting.
when I fired the test from link:
http://tests.ckeditor.dev:1030/#tests/path:/tests/plugins/indentlist/manual with a play button it seemd to be ignored, but there is no notification when test is opened directly.

I moved condition to body to work properly.

**Pull request merged**